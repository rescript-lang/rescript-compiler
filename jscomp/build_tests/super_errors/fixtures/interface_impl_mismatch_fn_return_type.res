module X: {
  type x = {
    name: string,
    age: int,
  }
  let make: int => string
} = {
  type x = {name: string, age: int}
  let make = s => {name: "", age: s}
}
