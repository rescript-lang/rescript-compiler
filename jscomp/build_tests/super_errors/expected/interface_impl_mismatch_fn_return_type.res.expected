
  [1;31mWe've found a bug for you![0m
  [36m/.../fixtures/interface_impl_mismatch_fn_return_type.res[0m:[2m7:5-10:1[0m

   5 [2mâ”‚[0m   }
   6 [2mâ”‚[0m   let make: int => string
   [1;31m7[0m [2mâ”‚[0m } = [1;31m{[0m
   [1;31m8[0m [2mâ”‚[0m [1;31m  type x = {name: string, age: int}[0m
   [1;31m9[0m [2mâ”‚[0m [1;31m  let make = s => {name: "", age: s}[0m
  [1;31m10[0m [2mâ”‚[0m [1;31m}[0m
  11 [2mâ”‚[0m 

  Signature mismatch:
  ...
  Value [1;33mmake[0m does not have the same type in the implementation and interface.
  
  Function returns type [1;31mx[0m in the implementation, but according to the interface it should be [1;33mstring[0m 
  
  [36m/.../fixtures/interface_impl_mismatch_fn_return_type.res[0m:[2m6:3-25[0m:
    Interface
  [36m/.../fixtures/interface_impl_mismatch_fn_return_type.res[0m:[2m9:7-10[0m:
    Implementation